### ISSUE TITLE
Layers of NATs and proxies

### ISSUE THREAD

**Issue by huitema** [quicwg/address-discovery#34]
From the minutes of QUIc at IETF 121

Lorenzo C: What if the Masque server does not have a public IP. Would you use STUN with this?

Marten: You could chain proxies… you would need to deal with this. Use of multiple nodes…

**Comment by marten-seemann** [comment]
There's nothing much we can do about this situation, and you have the exact same problem when using STUN.

Maybe we should point out that different endpoints might observe different addresses?

**Comment by marten-seemann** [comment]
@huitema Do you think we need to do anything here?

**Comment by huitema** [comment]
I don't think we need to do anything in particular. We are discovering a "reflexive" address mapping, which indeed depends on the routing, layers of NAT, etc. I think this is fixed by PR #47.
